### DISCUSSION TITLE
Does Prisma allow selecting data from nested relations?

### DISCUSSION THREAD

**Discussion by vsdepontes** [prisma/prisma#12890] [Q&A] [answered]
I am having a problem with TypeORM and I am seriously considering migrating my application to Prisma due to its benefits. One major limitation for me in TypeORM is basically selecting specific properties (columns) from nested relations, instead of getting all of them and having to remove later. I described my problem with an example here: [TypeORM select data from nested relations](https://stackoverflow.com/questions/71928918/typeorm-select-data-from-nested-relations)
How could I achieve that in Prisma?

**Comment by nurul3101** [comment, marked as answer]
Hey @vsdepontes 👋 

Considering this example schema
```prisma
model User {
  id           Int              @id @default(autoincrement())
  name         String?
  email        String           @unique
  profileViews Int              @default(0)
  role         Role             @default(USER)
  coinflips    Boolean[]
  posts        Post[]
  profile      ExtendedProfile?
}

model Post {
  id         Int        @id @default(autoincrement())
  title      String
  published  Boolean    @default(true)
  author     User       @relation(fields: [authorId], references: [id])
  authorId   Int
  comments   Json?
  views      Int        @default(0)
  likes      Int        @default(0)
  categories Category[]
}
```
Here's how you could use `select` on nested relations, In this case from the `Post` model.

```javascript
const users = await prisma.user.findMany({
  select: {
    name: true,
    posts: {
      select: {
        title: true,
      },
    },
  },
})
```
The above query would just select `title` from the nested `posts` relation.

This is the sample response from above query

```json
{
   "name":"Prisma",
   "posts":[
      {
         "title":"PostgreSQL Database"
      },
      {
         "title":"All about databases"
      }
   ]
}
```

Here's a reference for using [select](https://www.prisma.io/docs/concepts/components/prisma-client/select-fields#select-specific-fields) clause.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
